<launch>
 <group ns="lidar1">
  <node name="lidar01" pkg="bluesea2" type="bluesea2_node" output="screen" >
    #ROS#
    <param name="topic" value="scan"/>
    <param name="frame_id" value="map" />
     #DATA#
    <param name="raw_bytes" value="3"/>
    <param name="min_dist" value="0.01"/>
    <param name="max_dist" value="50.0"/>
    <param name="from_zero" value="false"/>
    <param name="output_scan" value="true" />
    <param name="output_cloud" value="false"/>
    <param name="output_360" value="true" />
    <param name="inverted" value="true"/>
    <param name="reversed" value="false"/>
    <param name="hard_resample" value="false"/>
    <param name="soft_resample" value="false"/>
    <param name="with_angle_filter" value="false"/>
    <param name="min_angle" value="-3.1415926"/>
    <param name="max_angle" value="3.1415926"/> 
    <rosparam param="mask1" >[-3.14,3.14]</rosparam>
    <!-- <rosparam param="mask2" >[-1,0]</rosparam> -->
    #CUSTOM#
    <param name="error_circle" value="3"/>#判断距离为0的点的比重      连续三圈
    <param name="error_scale" value="0.9"/>#每圈距离为0的点占据90%  则报错
    <param name="group_listener" value="false" />#仅用于监听组播数据(需要上位机修改雷达的上传地址,并且固定上传，然后该驱动监听数据)
    <param name="group_ip" value="224.0.0.11" />#监听的组播ip
    #FITTER#
    <param name="filter_open" value="true"/>#滤波生效开关
    <param name="max_range" value="20"/> #滤波生效的最大距离
    <param name="min_range" value="0.5"/>#滤波生效的最小距离
    <param name="max_range_difference" value="0.1"/>#离异点判断的物理范围
    <param name="filter_window" value="1"/>#判断离异点下标的范围
    #CONNECT#
    <param name="type" value="udp" />
    <param name="lidar_port" value="6543" />
    <param name="local_port" value="6668" />
    <param name="lidar_ip" value="192.168.158.98"/>
    #GET#
    <param name="uuid" value="-1" />
    #SET#
    <param name="rpm" value="-1"/> #-1  600 900  ...
    <param name="resample_res" value="-1"/>#-1 0 1
    <param name="with_smooth" value="-1" />#-1 0 1
    <param name="with_deshadow" value="-1" />#-1 0 1
    <param name="alarm_msg" value="-1"/>#-1 0 1
    <param name="direction" value="-1"/>#-1 0 1
  </node>
 </group>
</launch>